src/perception/filter.hpp

```diff
@@ -88,4 +90,21 @@ class MeanMedianFilter {
         auto end = mSortedValues.end() - (mProportion * size() / 4);
         return std::accumulate(begin, end, T{}) / (end - begin);
     }
+};
+
+/**
+ * @brief Combined filter for XYZ coordinates. Type is always double.
+ */
+struct XYZFilter {
```

Comment:
I'm not going to absolutely insist on this right now but really this should've been written with Eigen so that you could template the filter on arbitrarily sized vector inputs on a per-element basis. Maybe open a ticket as another low pri refactor.

Reply:
Yeah it's not used

Reply:
Can refactor it later if we need it